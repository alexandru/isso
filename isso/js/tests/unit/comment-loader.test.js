/**
 * @jest-environment jsdom
 */

/* Keep the above exactly as-is!
 * https://jestjs.io/docs/configuration#testenvironment-string
 * https://jestjs.io/docs/configuration#testenvironmentoptions-object
 */

"use strict";

test('Create comment loader', () => {
  // Set up our document body
  document.body.innerHTML =
    '<div id=isso-thread></div>' +
    // Note: `src` and `data-isso` need to be set,
    // else `api` fails to initialize!
    // data-isso-id needed for insert_loader api.fetch()
    '<script src="http://isso.api/js/embed.min.js"'
          + 'data-isso="/"'
          + 'data-isso-id="1"></script>';

  const isso = require("app/isso");
  const $ = require("app/dom");

  const config = require("app/config");
  const i18n = require("app/i18n");
  const svg = require("app/svg");

  const template = require("app/template");
  template.set("pluralize", i18n.pluralize);

  let comment = {
    'id': null,
    'hidden_replies': 5,
  }

  var isso_thread = $('#isso-thread');
  isso_thread.append('<div id="isso-root"></div>');

  isso.insert_loader(comment, null);

  // Will create a `.snap` file in `./__snapshots__/`.
  // Don't forget to check in those files when changing anything!
  expect(isso_thread.innerHTML).toMatchSnapshot();
});
