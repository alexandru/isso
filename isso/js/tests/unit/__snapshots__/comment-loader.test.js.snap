// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create comment loader 1`] = `"<div id="isso-root"><div class="isso-comment-loader" id="isso-loader-null"><a class="isso-load-hidden" href="#">5 Hidden</a></div></div>"`;
